core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1008,6 +975,10 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     }
   }
 
+  private def clearEventQueueAndReelect(): Unit = {
+    eventManager.clearAndPut(Reelect)
```

Comment:
If we don't try to trigger `elect` after we clear the queue and the new controller goes away before we clear the queue, the watch may have put `Reelect` in the queue before the clear happens. In this case, that broker will miss controller election. From the correctness point of view, this may not be a problem because at least one other broker will conduct the controller election and become the controller. It is safe to clear and resign if we don't care about fairness in controller election.

Reply:
Great point. We could just do Reelect here as you suggested. I was thinking that we could also potentially inline the logic (clear the event queue, mark the controller as inactive and call onControllerResignation()) here instead of enqueuing the logic to the event queue.  However, it seems that the former may be simpler.

Reply:
Agree. Code refactored.